umbraco/Umbraco-CMS
# Rollback improvements

### Prerequisites


If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/12119

### Description

This PR adds a bunch of improvements to the rollback dialog as suggested in #12119 

1. Never display the current draft version in the list of rollback options (it makes no sense, we can't rollback to the current draft version).
2. Always allow comparing to the currently published version - even if it is identical with the current draft version.
3. Introduce a "no changes" state to the comparison side of the rollback dialog. 
    - Note how this state disables the "rollback" button.
4. Remove the somewhat pointless "Current version" box above the list of available versions for rollback.
5. Some UI and textual updates to (hopefully) make things clearer/less confusing.

![rollback-improvements](https://user-images.githubusercontent.com/7405322/203144181-6cd809c0-331e-4d19-9d2d-7918bbe8020b.gif)

### Testing this PR

1. Ensure that you can perform a rollback to the previously published version when you have changes to the current draft.
2. Ensure that you are met by the new "no changes" state when there are no changes between the currently published version and the current draft.
3. Make sure you can perform rollbacks further backwards in the version history (provided there are changes, of course).
